cfme/tests/openstack/cloud/test_instances.py

```diff
@@ -0,0 +1,151 @@
+"""Tests for Openstack cloud instances"""
+
+import fauxfactory
+import pytest
+
+from cfme.cloud.instance.openstack import OpenStackInstance
+from cfme.cloud.provider.openstack import OpenStackProvider
+from cfme.web_ui import Quadicon
+from utils import testgen
+from utils.appliance.implementations.ui import navigate_to
+from utils.version import current_version
+
+pytest_generate_tests = testgen.generate([OpenStackProvider],
+                                         scope='module')
+
+pytestmark = [pytest.mark.usefixtures("setup_provider_modscope")]
+
+
+@pytest.fixture(scope='function')
+def new_instance(provider):
+    prov_data = provider.get_yaml_data()['provisioning']
+    instance = OpenStackInstance(fauxfactory.gen_alpha(), provider,
+                                 template_name=prov_data['image']['name'])
+    navigate_to(instance, 'Provision')
+    instance.create(fauxfactory.gen_email(), fauxfactory.gen_alpha(),
+                    fauxfactory.gen_alpha(), prov_data['cloud_network'],
+                    prov_data['instance_type'], False,
+                    security_groups='default',
+                    availability_zone=prov_data['availability_zone'],
+                    cloud_tenant=prov_data['tenant'])
+    instance.wait_to_appear()
+    return instance
+
+
+@pytest.fixture(scope='function')
+def shelved_instance(new_instance):
+    new_instance.power_control_from_provider(OpenStackInstance.SHELVE)
+    navigate_to(new_instance, 'Details')
+    new_instance.wait_for_instance_state_change(OpenStackInstance.STATE_SHELVED)
+    return new_instance
+
+
+def test_create_instance(new_instance, soft_assert):
+    """Creates an instance and verifies it appears on UI"""
+    navigate_to(new_instance, 'Details')
+    prov_data = new_instance.provider.get_yaml_data()['provisioning']
+    power_state = new_instance.get_detail(properties=('Power Management',
+                                                      'Power State'))
+    assert power_state == OpenStackInstance.STATE_ON
+
+    vm_tmplt = new_instance.get_detail(properties=('Relationships',
+                                                   'VM Template'))
+    soft_assert(vm_tmplt == prov_data['image']['name'])
+
+    # Assert other relationships in a loop
+    props = [('Availability Zone', 'availability_zone'),
+             ('Cloud Tenants', 'tenant'),
+             ('Flavor', 'instance_type')]
+
+    if current_version() >= '5.7':
+        props.append(('Virtual Private Cloud', 'cloud_network'))
+
+    for p in props:
+        v = new_instance.get_detail(properties=('Relationships', p[0]))
+        soft_assert(v == prov_data[p[1]])
+
+
```

Comment:
Okay, that's fine for now, just wanted to make sure you were aware of the existing tests. We're working on solving this at a higher level to reduce duplication across integration_tests and any eventual repo splits.